core/src/test/java/jenkins/monitor/OperatingSystemEndOfLifeAdminMonitorTest.java

```diff
@@ -0,0 +1,334 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package jenkins.monitor;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.not;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertThrows;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Random;
+import org.junit.Test;
+
+public class OperatingSystemEndOfLifeAdminMonitorTest {
+
+    private final OperatingSystemEndOfLifeAdminMonitor monitor;
+    private final Random random = new Random();
+    private final String PREFIX = "administrativeMonitor/";
+
+    public OperatingSystemEndOfLifeAdminMonitorTest() throws IOException {
+        this.monitor = random.nextBoolean()
+                ? new OperatingSystemEndOfLifeAdminMonitor()
+                : new OperatingSystemEndOfLifeAdminMonitor(OperatingSystemEndOfLifeAdminMonitor.class.getName());
+    }
+
+    @Test
+    public void testGetDisplayName() {
+        assertThat(monitor.getDisplayName(), is("Operating system end of life monitor"));
+    }
+
+    @Test
+    public void testGetAfterEndOfLifeDate() {
+        assertFalse(monitor.getAfterEndOfLifeDate());
+    }
+
+    @Test
+    public void testGetDocumentationUrl() {
+        assertThat(monitor.getDocumentationUrl(), is(not(nullValue())));
+    }
+
+    @Test
+    public void testGetEndOfLifeDate() {
+        assertThat(monitor.getEndOfLifeDate(), is("2099-12-31"));
+    }
+
+    @Test
+    public void testGetOperatingSystemName() {
+        /* Operating system name depends on the operating system running test */
+        assertThat(monitor.getOperatingSystemName(), not(nullValue()));
+    }
+
+    @Test
+    public void testGetSearchUrl() {
+        assertThat(monitor.getSearchUrl(), is(PREFIX + monitor.getClass().getName()));
+    }
+
+    @Test
+    public void testGetUrl() {
+        assertThat(monitor.getUrl(), is(PREFIX + monitor.getClass().getName()));
+    }
+
+    @Test
+    public void testIsActivated() throws IOException {
+        // Will fail if operating system running the test is reaching end of life soon
+        assertFalse(monitor.isActivated());
+    }
+
+    @Test
+    public void testIsSecurity() throws IOException {
+        assertFalse(monitor.isSecurity());
+    }
+
+    @Test
+    public void testNotIsActivatedWhenIgnoreEndOfLife() throws IOException {
+        monitor.ignoreEndOfLife = true;
+        assertFalse(monitor.isActivated());
+    }
+
+    private String docsUrl(String component) {
+        return "https://www.jenkins.io/redirect/operating-system-end-of-life?q=" + component + "#" + component;
```

Comment:
why are we using a query string and a fragment?

Reply:
Fragment can be used to redirect as is done with the "Submit an issue" redirect.  Query string is used in case there is a way to get metrics from it.

I'm fine any combination of them.  Let me know which you prefer and I'll make it that way.

The base URL is enough for my needs currently.  The base URL plus fragment is enough for the existing use cases of www.jenkins.io redirects.  The base URL plus fragment plus query string was a wild guess that it might be useful someday.

Reply:
Why can fragment be used for redirect and query string can't?
i.e. doesn't query string do both analytics and the ability to redirect to a specific page later if we wish

Reply:
https://github.com/jenkins-infra/jenkins.io/blob/master/content/participate/report-issue/redirect.html.haml uses the fragment so that https://www.jenkins.io/participate/report-issue/redirect/#15543 opens a page that will submit https://issues.jenkins.io/secure/CreateIssueDetails!init.jspa?pid=10172&issuetype=1&priority=4&components=15543 so that the bug report includes the correct component.

The fragment is extracted from the URL in the JavaScript in that page so that it can be included in the URL submitted to Jira.  I wasn't sure if there is a way to do the same thing with a query string

Reply:
Yes you can get it with: `new URLSearchParams(window.location.search).get('q')`

Reply:
Thanks.  I'll remove the fragment so that the code only passes the query parameter

Reply:
Removed in 0a35a4a1db5ddad106e01f63cf48109e7e5028b3